cristin-person/src/test/java/no/unit/nva/cristin/person/update/CristinPersonPatchJsonCreatorTest.java

```diff
@@ -1,96 +1,46 @@
-import static no.unit.nva.testutils.RandomDataGenerator.randomUri;
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import com.fasterxml.jackson.core.JsonProcessingException;
-import java.util.Arrays;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.nio.file.Path;
 import java.util.Collections;
-import no.unit.nva.cristin.model.CristinTypedLabel;
-import no.unit.nva.cristin.person.model.cristin.CristinPersonEmployment;
-import no.unit.nva.cristin.person.model.nva.PersonNvi;
-import no.unit.nva.cristin.person.model.nva.TypedValue;
-import nva.commons.apigateway.exceptions.BadRequestException;
-import nva.commons.core.paths.UriWrapper;
+import nva.commons.core.ioutils.IoUtils;
 import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
 
 public class CristinPersonPatchJsonCreatorTest {
 
-    private static final String VALID_ORCID = "1234-1234-1234-1234";
-    private static final String SOME_NAME = randomString();
-    private static final String SOME_OTHER_NAME = randomString();
-    private static final String SOME_PREFERRED_NAME = randomString();
-    public static final String END_OF_LITERAL = "\" , ";
-    public static final String ENGLISH_LANG = "en";
-    public static final String NORWEGIAN_LANG = "no";
-    public static final String NORWEGIAN_LANG_CONTENT = "Norsk";
-    public static final String SOME_NUMBER = "11223344";
-    public static final String SOME_EMAIL = "hello@example.org";
-    public static final String SOME_WEBPAGE = "www.example.org";
-
     @Test
-    void shouldParseInputJsonIntoCristinJsonWithCorrectMappingOfFields() {
-        var input = OBJECT_MAPPER.createObjectNode();
-        input.put(ORCID, VALID_ORCID);
-        input.put(FIRST_NAME, SOME_NAME);
-        input.put(LAST_NAME, SOME_OTHER_NAME);
-        input.put(PREFERRED_FIRST_NAME, SOME_PREFERRED_NAME);
-        input.putNull(PREFERRED_LAST_NAME);
-        input.put(RESERVED, true);
-
-        var contactDetails = OBJECT_MAPPER.createObjectNode();
-        contactDetails.put(TELEPHONE, SOME_NUMBER);
-        contactDetails.put(EMAIL, SOME_EMAIL);
-        contactDetails.put(WEB_PAGE, SOME_WEBPAGE);
+    void shouldParseInputJsonIntoCristinJsonWithCorrectMappingOfFieldsAsAuthorizedClient() throws Exception {
+        var inputString = IoUtils.stringFromResources(Path.of("nvaApiUpdatePersonAsAdminRequest.json"));
+        var input = (ObjectNode) OBJECT_MAPPER.readTree(inputString);
+        var result = new CristinPersonPatchJsonCreator(input).create().getOutput();
+        var expectedString = IoUtils.stringFromResources(Path.of("cristinUpdatePersonAsAdminRequest.json"));
 
-        input.set(CONTACT_DETAILS, contactDetails);
+        JSONAssert.assertEquals(expectedString, result.toString(), JSONCompareMode.NON_EXTENSIBLE);
```

Comment:
Because if any fields is missing from the model class and we do a compare, those fields are not accounted for and we might get a false positive. Jsonassert compares the fields from json directly.